prow/moonraker/client.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package moonraker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+)
+
+type Client struct {
+	host       string
+	httpClient *http.Client
+}
+
+func NewClient(host string, timeout time.Duration) *Client {
+	return &Client{
+		host: host,
+		httpClient: &http.Client{
+			Timeout: timeout,
+		},
+	}
+}
+
+func (c *Client) do(method, endpoint string, payload []byte, params map[string]string) (*http.Response, error) {
+	baseURL := fmt.Sprintf("%s/%s", c.host, endpoint)
```

Comment:
Please use `url.JoinPath`.

nit: `path` is probably more precise terminology than `endpoint` in this function.